pkg/controller/container-runtime-config/helpers.go

```diff
@@ -933,17 +933,69 @@ func validateClusterImagePolicyWithAllowedBlockedRegistries(clusterScopePolicies
 	return nil
 }
 
-func generateSigstoreRegistriesdConfig(clusterScopePolicies map[string]signature.PolicyRequirements) ([]byte, error) {
+func generateSigstoreRegistriesdConfig(clusterScopePolicies map[string]signature.PolicyRequirements, registriesTOML []byte) ([]byte, error) {
 	if len(clusterScopePolicies) == 0 {
 		return nil, nil
 	}
 
-	registriesDockerConfig := make(map[string]dockerConfig)
-	sigstoreAttachment := dockerConfig{
-		UseSigstoreAttachments: true,
+	var (
+		err                    error
+		tmpFile                *os.File
+		registriesConf         = sysregistriesv2.V2RegistriesConf{}
+		registriesDockerConfig = make(map[string]dockerConfig)
+		sigstoreAttachment     = dockerConfig{
+			UseSigstoreAttachments: true,
+		}
+	)
+
+	if registriesTOML != nil {
+		tmpFile, err = os.CreateTemp("", "regtemp")
+		if err != nil {
+			return nil, err
+		}
+		defer func() {
+			tmpFile.Close()
+			os.Remove(tmpFile.Name())
+		}()
+
+		_, err = tmpFile.Write(registriesTOML)
+		if err != nil {
+			return nil, err
+		}
+
+		if _, err := toml.NewDecoder(bytes.NewBuffer(registriesTOML)).Decode(&registriesConf); err != nil {
+			return nil, fmt.Errorf("error decoding registries config: %w", err)
+		}
 	}
-	for scope := range clusterScopePolicies {
-		registriesDockerConfig[scope] = sigstoreAttachment
+
+	for policyScope := range clusterScopePolicies {
+		registriesDockerConfig[policyScope] = sigstoreAttachment
+		if registriesTOML == nil {
+			continue
+		}
+		parentReg, err := sysregistriesv2.FindRegistry(&types.SystemContext{SystemRegistriesConfPath: tmpFile.Name()}, policyScope)
```

Comment:
I think this should set `SystemRegistriesConfDirPath` to something non-existent or not-a-directory, so that the controller doesn’t read current configuration of the node/container where it is currently running.

(In both places, ideally by sharing the `SystemContext`)

Reply:
Thanks for the explanation.